Sources/TuistLoader/Models+ManifestMappers/Config+ManifestMapper.swift

```diff
@@ -36,49 +36,46 @@ extension TuistCore.Config {
         rootDirectory: AbsolutePath,
         at path: AbsolutePath
     ) async throws -> TuistCore.Config {
-        let generatorPaths = GeneratorPaths(manifestDirectory: path, rootDirectory: rootDirectory)
-        var generationOptions = try TuistCore.Config.GenerationOptions.from(
-            manifest: manifest.generationOptions,
-            generatorPaths: generatorPaths
-        )
-        let compatibleXcodeVersions = TuistCore.CompatibleXcodeVersions.from(manifest: manifest.compatibleXcodeVersions)
-        let plugins = try manifest.plugins.map { try PluginLocation.from(manifest: $0, generatorPaths: generatorPaths) }
-        let swiftVersion: TSCUtility.Version?
-        if let configuredVersion = manifest.swiftVersion {
-            swiftVersion = TSCUtility.Version(configuredVersion.major, configuredVersion.minor, configuredVersion.patch)
-        } else {
-            swiftVersion = nil
-        }
+        switch manifest.project {
+        case let .tuist(compatibleXcodeVersions, manifestSwiftVersion, plugins, generationOptions, installOptions):
+            let generatorPaths = GeneratorPaths(manifestDirectory: path, rootDirectory: rootDirectory)
+            var generationOptions = try TuistCore.Config.GenerationOptions.from(
```

Comment:
```suggestion
            let generationOptions = try TuistCore.Config.GenerationOptions.from(
```

I don't see this being mutated anywhere?